client/src/components/SubmissionForm.jsx

```diff
@@ -107,7 +107,7 @@ export default function SubmissionForm() {
                 {/* eslint-disable-next-line */}
                 {values.links.map((link, index) => {
                   return (
-                    <div key={link.label}>
```

Comment:
Hm okay removing the key param is probably not the best way to fix this, but its interesting that it does work. Maybe check if the keys are unique? Or maybe try to find another fix that doesnt involve removing the key (youll probably see a warning in the console when you run the app after removing the key)